src/ciphersuite/mod.rs

```diff
@@ -395,14 +407,55 @@ impl Ciphersuite {
 }
 
 impl AeadKey {
-    /// Build a new key for an AEAD from `Secret`.
-    pub(crate) fn from_secret(secret: Secret, aead_mode: AeadMode) -> AeadKey {
+    /// Create an `AeadKey` from a `Secret`. TODO: This function should
+    /// disappear when tackling issue #103.
+    pub(crate) fn from_secret(ciphersuite: &Ciphersuite, secret: Secret) -> Self {
         AeadKey {
-            aead_mode,
+            aead_mode: ciphersuite.aead,
             value: secret.value,
         }
     }
 
+    fn expand_from_secret(ciphersuite: &Ciphersuite, ciphertext: &[u8], secret: &Secret) -> Self {
```

Comment:
Hm, why did you pull this out if it's only used in one function? Did you want to re-use it in `from_welcome_secret`?

Reply:
Originally, yes. I think it's going to be used when deriving the AeasKeys from the SecretTree. Not sure, yet, though. I'll put it back in.